### ISSUE TITLE
Segment the cell and the nucleus simultaneously

### ISSUE THREAD

**Issue by XiangruiJi** [SchmollerLab/Cell_ACDC#165]
I want to segment cells (bright-field channel) using YeaZ and the nuclei (fluorescence channel) using StarDist at the same time and give the cell and its nucleus the same ID. Also, I want to eliminate the nucleus without a corresponding cell, because that nucleus may not be a real one. But now, although I load the bright-field channel and overlay the fluorescence channel, I can only do segmentation for the bright-field channel.

I wish you could add this function. This function will be helpful in calculating the nucleus-cell ratio or the amount of some nucleus-localized proteins. Thanks!

**Comment by ElpadoCan** [comment]
Hi @XiangruiJi, that is a great idea! I will add a utility for that soon.

Note that you can already segment as many channels as you want. If, for example, you first segment the bright field signal and then on the same position you repeat segmentation and choose the nuclear signal then Cell-ACDC will ask you to append a name to the segmentation file. Let's suppose you enter the name "nucleus". Then, Cell-ACDC will save a segmentation file ending with `segm_nucleus.npz` with a connected features file called `acdc_output_nucleus.csv`. This csv file will contain numerical features of the nucleus signal like volume, area, intensities etc. As I said you can repeat this for as many channels as you want.

However, you are not guaranteed that nuclei will have the same IDs of the cells they belong to, so that is what the new utility will solve.

Long story short, would an utility that "assigns to the nuclei the same IDs of the cells they belong to" be a solution for your situation?

Cheers, 
Francesco

**Comment by XiangruiJi** [comment]
Yeah, this utility is just what I want. If it comes true, I can first segment and track the cells and correct segmentation and tracking errors. Then, I can segment the nuclei and use this utility to assign IDs to them (If a nucleus belongs to no cell, it will be deleted).

Thanks a lot!

**Comment by XiangruiJi** [comment]
Hi @ElpadoCan, I note that you've create this utility #168, but I fail to find this utility in the latest version of the software. Where can I find it? Thanks!

**Comment by ElpadoCan** [comment]
Hi @XiangruiJi, you are right, we have that utility now, but it's only on the developer version. I didn't release it yet because we are still testing it out. If you want to try it right now, you need to install the developer version of Cell-ACDC, which means you need to clone the GitHub repo and then install the cloned version.

It's actually not that complicated, let me know if you are interested and I can give you instructions. Otherwise, you could wait a couple of days and I will release a new version with that utility. Thank you for your patience! 

Cheers,
Francesco

**Comment by XiangruiJi** [comment]
OK, I can wait. Looking forward to the new version!

**Comment by ElpadoCan** [comment]
Hi @XiangruiJi I released `v1.2.4rc46` with the utility to track the sub-cellular objects (i.e., assigning to the sub-cellular objects the same ID as the cell they belong to).

To use it, first create two segmentation files, one for the cells and one for the sub-cellular objects (e.g., nucleus).

Then, run the utility from the `Utilities` menu on the top menubar of the little launcher that opens when you launch acdc. The utility is called `Track sub-cellular objects (assign same ID as the cell they belong to)....`

Finally, follow the pop-up windows, hopefully they are self-explanatory. If they are not (or you have quesitons on how it works), do not hesitate to comment here, or under issue #168.

Please, update with `pip install --upgrade cellacdc` (after activating the environment) and let me know if it works, thanks!

Cheers,
Francesco

**Comment by XiangruiJi** [comment]
It does work. Thanks for your hard work!

**Comment by ElpadoCan** [comment]
Great, thanks for letting me know! 

I will close this issue now but fee free to reopen it if you have question or problems.

Cheers
Francesco
